packages/js/product-editor/src/blocks/name/edit.tsx

```diff
@@ -136,6 +138,16 @@ export function Edit() {
 		'product_name'
 	) as string;
 
+	const inputRef = useRef< HTMLInputElement >( null );
+
+	useEffect( () => {
+		if ( inputRef.current ) {
+			inputRef.current.focus();
```

Comment:
I don't think this should be the default for the `name` block, as it could very well be used in other contexts where the auto focus wouldn't make sense.
I think we are better off adding this as an option through attributes. We may want a `autoFocus` attribute. You can see the radio block for examples on using additional attributes, and how to get the value within the component.

I would also suggest using the `autoFocus` prop for the `inputControl` instead of going the above route.

Reply:
Thanks for the suggestion. I didn't knew that there was an `autoFocus` prop since we don't have the type definitions :(